ichnaea/conf.py

```diff
@@ -0,0 +1,117 @@
+"""
+Application settings and configuration derived from the environment
+"""
+from __future__ import absolute_import
+
+import os
+import os.path
+
+from everett.component import ConfigOptions, RequiredConfigMixin
+from everett.manager import ConfigManager, ConfigOSEnv
+
+HERE = os.path.dirname(__file__)
+
+
+class AppConfig(RequiredConfigMixin):
+    required_config = ConfigOptions()
+    required_config.add_option(
+        'testing',
+        default='false',
+        parser=bool,
+        doc='Whether or not we are running tests.'
+    )
+
+    required_config.add_option(
+        'asset_bucket',
+        default='',
+        doc='name of AWS S3 bucket to store map tile image assets in'
+    )
+    required_config.add_option(
+        'asset_url',
+        default='',
+        doc='Amazon CloudFront url for map tile image assets'
+    )
+
+    # Database related settings
+    required_config.add_option(
+        'db_library',
+        default='pymysql',
+        doc='one of "pymysql" or "mysqlconnector"'
+    )
+    required_config.add_option(
+        'db_readonly_uri',
+        doc=(
+            'uri for the readonly database; '
+            'mysql+LIBRARY://USER:PASSWORD@HOST:PORT/NAME'
+        )
+    )
+    required_config.add_option(
+        'db_readwrite_uri',
+        doc=(
+            'uri for the read-write database; '
+            'mysql+LIBRARY://USER:PASSWORD@HOST:PORT/NAME'
+        )
+    )
+    required_config.add_option(
+        'db_ddl_uri',
+        doc=(
+            'uri for the ddl database used for migrations; '
+            'mysql+LIBRARY://USER:PASSWORD@HOST:PORT/NAME'
+        )
+    )
+
+    required_config.add_option(
+        'sentry_dsn',
+        default='',
+        doc='Sentry DSN; leave blank to disable Sentry error reporting'
+    )
+
+    required_config.add_option(
+        'statsd_host',
+        default='',
+        doc='StatsD host; blank to disable StatsD'
+    )
+
+    required_config.add_option(
+        'redis_uri',
+        doc='uri for Redis; redis://HOST:PORT/DB'
+    )
+
+    required_config.add_option(
+        'mapbox_token',
+        default='',
+        doc=(
+            'Mapbox API key; if you do not provide this, then parts of '
+            'the site will not work'
```

Comment:
For today, this is the right code, and good job including the significance of leaving it at the default. 👍

I'd like to revisit this soon. When unset, the website is a placeholder, and when set it looks like https://location.services.mozilla.com/. I'd rather always see something that looks like the location website, with a message like "map disabled, no Mapbox API key specified", to ease development. We'll see if anyone comes forward as users of [ichnaea as a platform](https://groups.google.com/forum/#!topic/mozilla.dev.geolocation/pWB4LQ3UyrM).

Reply:
I like that idea, too. Seems weird to shut off the entire website if there's no mapbox api key plus it'd make it easier to develop if it didn't do that.